MediaBrowser.Api/Playback/Hls/DynamicHlsService.cs

```diff
@@ -722,22 +722,155 @@ private bool EnableAdaptiveBitrateStreaming(StreamState state, bool isLiveStream
+            {
+                return HLSCodecStringFactory.GetMp3String();
+            }
+            else if (string.Equals(state.ActualOutputAudioCodec, "ac3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return HLSCodecStringFactory.GetAC3String();
+            }
+            else if (string.Equals(state.ActualOutputAudioCodec, "eac3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return HLSCodecStringFactory.GetEAC3String();
+            }
+            return string.Empty;
+        }
+
+        private string GetPlaylistVideoCodecs(StreamState state)
         {
-            var header = "#EXT-X-STREAM-INF:BANDWIDTH=" + bitrate.ToString(CultureInfo.InvariantCulture) + ",AVERAGE-BANDWIDTH=" + bitrate.ToString(CultureInfo.InvariantCulture);
+            int level = Convert.ToInt32(state.GetRequestedLevel(state.ActualOutputVideoCodec));
 
-            // tvos wants resolution, codecs, framerate
-            //if (state.TargetFramerate.HasValue)
-            //{
-            //    header += string.Format(",FRAME-RATE=\"{0}\"", state.TargetFramerate.Value.ToString(CultureInfo.InvariantCulture));
-            //}
+            if (string.Equals(state.ActualOutputVideoCodec, "h264", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string profile = state.GetRequestedProfiles("h264").FirstOrDefault();
+                return HLSCodecStringFactory.GetH264String(profile, level);
+            }
+            else if (string.Equals(state.ActualOutputVideoCodec, "h265", StringComparison.InvariantCultureIgnoreCase)
+                    || string.Equals(state.ActualOutputVideoCodec.ToLower() == "hevc", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string profile = state.GetRequestedProfiles("h265").FirstOrDefault();
+                return HLSCodecStringFactory.GetH265String(profile, level);
+            }
+            return string.Empty;
+        }
+
+        private void AppendPlaylistCodecsField (StringBuilder field, StreamState state)
+        {
+            // Video
+            string videoCodecs = string.Empty;
+            if (!string.IsNullOrEmpty(state.ActualOutputVideoCodec))
+            {
+                videoCodecs = GetPlaylistVideoCodecs(state);
+            }
+
+            // Audio
+            string audioCodecs = string.Empty;
+            if (!string.IsNullOrEmpty(state.ActualOutputAudioCodec))
+            {
+                audioCodecs = GetPlaylistAudioCodecs(state);
+            }
+
+            if (!string.IsNullOrEmpty(videoCodecs) || !string.IsNullOrEmpty(audioCodecs))
+            {
+                field.Append(",CODECS=\"");
+
+                if (!string.IsNullOrEmpty(videoCodecs) && !string.IsNullOrEmpty(audioCodecs))
+                {
+                    // If both video and audio exists
+                    field.Append(videoCodecs + "," + audioCodecs);
```

Comment:
Shouldn't it be like this?
```suggestion
                    field.Append(videoCodecs);
                    field.Append(",");
                    field.Append(audioCodecs);
```

Reply:
Yes, and you can chain multiple `Appends`